### ISSUE TITLE
[feature request]

### ISSUE THREAD

**Issue by odlagi-t** [zombieFox/nightTab#201]
Hi, this project is very nice and helpful...but maybe from a different angle a problem-solver !

Some parents, with young-age kids, are aware of landing-page of the www.
I find it preferable having some view to web-stuff my kid can potentially visit.
The covid impacted regular teaching , to primary distance home-education.
Web content to digest is with incremental steps.

What about follow suggestions ?
Option to discard the google search bar,but looking up bookmarks is fine, and
as a consequence discard the 'setting' knob. 
And add-on is an open-closing sliding window, when moving cursor to the very left or right of screen.
Here we, me and partner could make a small to-do list, highlighting points/subjects...etc for the week/month.
We could preset the bookmarks related to her growing interests. And import as a startpage on her computer?

Thank you. 
Nice work you have done !
I found this project via reddit-post.

**Comment by zombieFox** [comment]
> Hi, this project is very nice and helpful...but maybe from a different angle a problem-solver !

Hi, thank you. I'll try my best to help with but I'm sorry I don't think I understand some details of this feature request.

> Option to discard the google search bar,but looking up bookmarks is fine

It is possible to completely  hide the search bar from the settings menu. But, do you mean showing the search bar to search for bookmarks but not using a search engine when pressing the enter key?

> ... discard the 'setting' knob.

Do you mean hiding the settings button? If so I'm afraid that would make accessing the settings menu impossible once hidden. The settings menu button is the explicit means to gain access to the controls so hiding it would negatively impact the user experience. 

> And add-on is an open-closing sliding window, when moving cursor to the very left or right of screen.
Here we, me and partner could make a small to-do list, highlighting points/subjects...etc for the week/month.

I've given this some thought in the past and I do not want to build note taking/to-do list features in this project. From previous experience building such features will add more complexity to the project than I am able or willing to support. I am sorry but I am aiming to keep the scope of this project to bookmarks only. 

> We could preset the bookmarks related to her growing interests. And import as a startpage on her computer?

I do apologies, I do not understand this question very well. Do you mean importing bookmarks from another browser into nightTab?

**Comment by odlagi-t** [comment]
Very kind from you giving a reply. Sorry, but not native English & poor it knowledge.
 

> but not using a search engine when pressing the enter key?

yes exactly.... I see if in case this could be an option...is hit 'ENTER' without results.
 

> is the explicit means to gain access to the controls (setting knob)

...I understand. But, here in 'problem-solve' case, this would be the key....else she can try cheating behind my back.(see **)

> project to bookmarks only

I do understand. A slider with static-text...We can write on paper! But, hmm what about dynamic...RSS feed (meteo,F1,football,etc)

 

> importing bookmarks from

I try to clarify. I tune her 'startpage' on my side. I do 'export'. And on her device I import the file.
** if 'locked' export then the imported nighTab maybe with hidden setting knob.

**Comment by Excigma** [comment]
Perhaps there can be a hidden setting in localstorage to hide the setting icon, that is not available in the settings page itself, however this might become complicated as more "hidden features" get added...

Storing "keys" (passwords) securely will be another problem as everything is client side + open source.

Locking access to the web may be difficult using only the newtab page, your child will still be able to type URLs (eg https://google.com) into the url bar to go to websites directly..

For your note-taking needs a (bad) idea is to use the hotkey `Ctrl` + `Alt` + `A` to create a bookmark with no url and the note content as the title and add it to a Notes Group, then deleting the bookmark when done.

A better way in my opinion is to whitelist sites that she can visit (using an extension on the browser, a program on the computer, or something on your wifi router), which this extension cannot do

**Comment by metruzanca** [comment]
> I try to clarify. I tune her 'startpage' on my side. I do 'export'. And on her device I import the file.

I'm not entirely sure how or if google chrome syncs the settings for this extension. If it does, you could probably accomplish this by having access to their google account and change the bookmarks then let google chrome handle things via sync.

**Comment by odlagi-t** [comment]
> For your note-taking needs a (bad) idea is to use the hotkey Ctrl + Alt + A to create a bookmark with no url and the note content as the title and add it to a Notes Group, then deleting the bookmark when done.

Indeed Excigma, I tried this 'note' effect. Is working w/ few short text-lines...

**Comment by zombieFox** [comment]
> I'm not entirely sure how or if google chrome syncs the settings for this extension. If it does, you could probably accomplish this by having access to their google account and change the bookmarks then let google chrome handle things via sync.

Yes, Chrome and Firefox do allow data to be synced between browsers assuming the user is logged in. See issue #93 for this feature request. It's still being investigated. It will require a new permission and I've tried hard to avoid needing user permissions.

**Comment by metruzanca** [comment]
> It will require a new permission and I've tried hard to avoid needing user permissions.

Any particular reason you're avoiding requesting user permissions?

**Comment by zombieFox** [comment]
Personal preference along with feedback and request from users on what permissions and data nightTab currently has access to. Seems people don't like extensions/addons with too many permissions. 

This was an interesting read on the topic: https://www.reddit.com/r/programming/comments/hi3tgf/i_lost_50000_chrome_extension_users_after_adding/

**Comment by metruzanca** [comment]
oof that OP lost 50k users. That's pretty rough. That article is quite interesting, though.
Its a shame there's no dynamic permission request system like android apps do (e.g. when you need to upload/select a file, it will ask permission to use storage). Currently you only get a bulk permission request on app install, though I wonder if that feature has at all been proposed for the browser extension API.

**Comment by zombieFox** [comment]
> dynamic permission request system like android apps ... though I wonder if that feature has at all been proposed for the browser extension API

It would be useful yes. Not sure either.
